src/taipy/core/common/_mongo_connector.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from typing import Dict, Optional
+
+import pymongo
+
+
+def _connect_mongodb(
```

Comment:
Oh we did not have the `@lru_cache` here? Can you copy the method from taipy-enterprise?

Reply:
As mentioned above, I tried to add lru and have another function as wrapper, but some tests passed while some failed and havent found the reason for that yet. Still investigating the issue but it also might not be as urgent since we didnt have it before and Im more concerned if it's blocking the release :D

Reply:
I share the same concern.

The purpose of the ticket is first to align the Pymongo versions. Using lru_cache is a bonus. If we can make it on time, that is great, but I would not postpone the release for that.

Can't we split the two topics into two PRs so that first we align the version and then we work on the lru_cache?

My two cents,

Reply:
I have created a ticket for the lru cache topic. Otherwise, I think the ticket is ready to be merged.

https://app.zenhub.com/workspaces/taipy-613ef328affde6000ff77fb3/issues/gh/avaiga/taipy-core/797

Reply:
Since Giang has already fixed the issue, I have closed the ticket